engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeConversionNode.java

```diff
@@ -196,6 +203,129 @@ Object doConvertText(
     }
   }
 
+  @Specialization(guards = {
+      "!typesLib.hasType(that)",
+      "!typesLib.hasSpecialDispatch(that)",
+      "!interop.isTime(that)",
+      "interop.isDate(that)",
+  })
+  Object doConvertDate(
```

Comment:
Conversions for builtins. Good. Shall we have a test that uses `ValuesGenerator` and tries conversion from each of the values?

Reply:
Sounds like a good idea. I will add these tests.